src/lib.rs

```diff
@@ -95,23 +106,33 @@ impl IRObject {
 
     pub fn get_metal_lib_binary(
         &self,
+        compiler: &IRCompiler,
         shader_stage: ffi::IRShaderStage,
-        dest_lib: &mut IRMetalLibBinary,
-    ) -> bool {
-        unsafe {
+    ) -> Result<IRMetalLibBinary, MetalLibNoBytecodeFoundError> {
+        let mtl_lib = IRMetalLibBinary::new(compiler);
+        if unsafe {
             self.funcs
-                .IRObjectGetMetalLibBinary(self.me, shader_stage, dest_lib.me)
+                .IRObjectGetMetalLibBinary(self.me, shader_stage, mtl_lib.me)
+        } {
+            Ok(mtl_lib)
+        } else {
+            Err(MetalLibNoBytecodeFoundError)
         }
     }
 
     pub fn get_reflection(
         &self,
+        compiler: &IRCompiler,
```

Comment:
Is an `IRObject` created from an `IRCompiler`? Maybe we could "somehow" store the `IRCompiler` inside `IRObject` so that:

1. It's less annoyance for the user;
2. Users can't pass the wrong compiler into an unrelated `IRObject`.

Reply:
I'm struggling to find a nice way to do that

Reply:
Solved via #14, merged into this PR.